cdap-api-common/src/main/java/io/cdap/cdap/api/data/format/StructuredRecord.java

```diff
@@ -192,7 +220,12 @@ public BigDecimal getDecimal(String fieldName) {
       return new BigDecimal(new BigInteger(Bytes.toBytes((ByteBuffer) value)), scale);
     }
 
-    return new BigDecimal(new BigInteger((byte[]) value), scale);
+    try {
+      return new BigDecimal(new BigInteger((byte[]) value), scale);
+    } catch (ClassCastException e) {
+      throw new UnexpectedFormatException(String.format("Field '%s' is expected to be a %s, but is a %s.", fieldName,
```

Comment:
Why do we retrow `ClassCastException` in all other cases, but `UnexpectedFormatException` here?

Reply:
good catch, not intended will change it